gazebo_ros/test/test_qos.cpp

```diff
@@ -0,0 +1,264 @@
+// Copyright 2020 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+#include <gazebo_ros/qos.hpp>
+
+TEST(TestQoS, ValidConstruction)
+{
+  // No SDF
+  {
+    gazebo_ros::QoS qos;
+  }
+  // No <qos> element
+  {
+    auto sdf_str =
+      "<?xml version='1.0' ?>"
+      "<sdf version='1.6'>"
+      "<world name='default'>"
+      "</world>"
+      "</sdf>";
+    sdf::SDF sdf;
+    sdf.SetFromString(sdf_str);
+    gazebo_ros::QoS qos(sdf.Root());
+
+    // We should get the default QoS for an arbitrary topic
+    rclcpp::QoS some_qos = qos.get_publisher_qos("some_random_test_topic");
+    EXPECT_EQ(rclcpp::QoS(10), some_qos);
+  }
+  // Empty <qos> element
+  {
+    auto sdf_str =
+      "<?xml version='1.0' ?>"
+      "<sdf version='1.6'>"
+      "<world name='default'>"
+      "<plugin name='node_1' filename='libnode_name.so'>"
+      "<qos />"
+      "</plugin>"
+      "</world>"
+      "</sdf>";
```

Comment:
nit: I would indent (using spaces in the strings) in these in line `xml` examples.
IMHO, it's more readable with indentation.

Reply:
Yeah, I'll try doing that. I was just following the style of the other tests in `gazebo_ros`.

Reply:
Done: 4ca784e